src/bun.js/bindings/JSBufferList.cpp

```diff
@@ -170,19 +192,23 @@ JSC::JSValue JSBufferList::_getBuffer(JSC::VM& vm, JSC::JSGlobalObject* lexicalG
     if (UNLIKELY(!array)) {
         return throwTypeError(lexicalGlobalObject, throwScope, "_getBuffer can only be called when all buffers are Uint8Array"_s);
     }
-    const size_t len = array->byteLength();
+    const size_t len = array->byteLength() - m_read;
     size_t n = total;
 
     if (n == len) {
         m_deque.removeFirst();
+        if (m_read > 0) {
+            auto buffer = array->possiblySharedBuffer();
+            array = JSC::JSUint8Array::create(lexicalGlobalObject, subclassStructure, buffer, m_read, len);
+            m_read = 0;
+        }
         RELEASE_AND_RETURN(throwScope, array);
     }
     if (n < len) {
         auto buffer = array->possiblySharedBuffer();
-        JSC::JSUint8Array* retArray = JSC::JSUint8Array::create(lexicalGlobalObject, subclassStructure, buffer, 0, n);
-        JSC::JSUint8Array* newArray = JSC::JSUint8Array::create(lexicalGlobalObject, subclassStructure, buffer, n, len - n);
-        iter->set(vm, this, newArray);
-        RELEASE_AND_RETURN(throwScope, retArray);
+        JSC::JSUint8Array* head = JSC::JSUint8Array::create(lexicalGlobalObject, subclassStructure, buffer, m_read, n);
```

Comment:
This needs a RETURN_IF_EXCEPTION. If the array is detached, it will throw (it needed it before too)

Reply:
https://github.com/oven-sh/bun/pull/1735/commits/e56d73e1d58cdb981d76d7f522bb02b49e2a33c1